src/DataStructures.jl

```diff
@@ -94,4 +94,6 @@ module DataStructures
     export status
     export deref_key, deref_value, deref, advance, regress
 
+    @deprecate iter(s::Stack) collect(s)
+    @deprecate iter(q::Queue) collect(q)
```

Comment:
I agree it's not unreasonable to break it given the intended use case. I think either way there are downsides, and it's one of those scenarios where you lose something either way. 

No strong feelings, so do what feels right.